forest/cli/src/cli/send_cmd.rs

```diff
@@ -7,9 +7,85 @@ use forest_json::message::json::MessageJson;
 use forest_rpc_client::{mpool_push_message, wallet_default_address};
 use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
 use num::BigInt;
+use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 
 use super::{handle_rpc_err, Config};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FILAmount {
+    pub value: TokenAmount,
+}
+
+impl FromStr for FILAmount {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let suffix_idx = s.rfind(char::is_numeric);
+        let (val, suffix) = match suffix_idx {
+            Some(idx) => s.split_at(idx + 1),
+            None => anyhow::bail!("failed to parse string: {}", s),
+        };
+
+        let mut multiplier = dec!(1.0);
+        let suffix = suffix.replacen(' ', "", 1).trim().to_lowercase();
+        match suffix.strip_suffix("fil").map(str::trim).unwrap_or(&suffix) {
+            "atto" | "a" => "atto",
+            "femto" => {
+                multiplier *= dec!(1e3);
+                "femto"
+            }
+            "pico" => {
+                multiplier *= dec!(1e6);
+                "pico"
+            }
+            "nano" => {
+                multiplier *= dec!(1e9);
+                "nano"
+            }
+            "micro" => {
+                multiplier *= dec!(1e12);
+                "micro"
+            }
+            "milli" => {
+                multiplier *= dec!(1e15);
+                "milli"
+            }
+            "" => {
+                multiplier *= dec!(1e18);
+                ""
+            }
+            _ => {
+                anyhow::bail!("unrecognized suffix: {}", suffix);
+            }
+        };
+
+        if val.chars().count() > 50 {
+            anyhow::bail!("string length too large: {}", val.chars().count());
+        }
+
+        let parsed_val = Decimal::from_str(val)
+            .map_err(|_| anyhow::anyhow!("failed to parse {} as a decimal number", val))?;
+
+        let attofil_val = if (parsed_val * multiplier).normalize().scale() != 0 {
+            anyhow::bail!("{} must convert to a whole attoFIL value", &s);
+        } else {
+            (parsed_val * multiplier).trunc().to_u128()
+        };
```

Comment:
You can refactor this code to be smaller. You can also change it to compute an `i128` instead of an `Option<i128>`.